store/commitment/iavl/tree.go

```diff
@@ -83,6 +94,142 @@ func (t *IavlTree) Prune(version uint64) error {
 	return t.tree.DeleteVersionsTo(int64(version))
 }
 
+// Snapshot implements snapshottypes.CommitSnapshotter.
+func (t *IavlTree) Snapshot(version uint64, protoWriter protoio.Writer) error {
+	if version == 0 {
+		return fmt.Errorf("the snapshot version must be greater than 0")
+	}
+
+	latestVersion := t.GetLatestVersion()
+	if version > latestVersion {
+		return fmt.Errorf("the snapshot version %d is greater than the latest version %d", version, latestVersion)
+	}
+
+	tree, err := t.tree.GetImmutable(int64(version))
+	if err != nil {
+		return fmt.Errorf("failed to get immutable tree for version %d: %w", version, err)
+	}
+
+	exporter, err := tree.Export()
```

Comment:
Could we please call this `itr` or `iter` instead of `exporter`? `exporter` makes it seem like its purpose is to write out content, but really the purpose is to iterate over the exported content.